src/core/ccitt_stream.js

```diff
@@ -0,0 +1,71 @@
+/* Copyright 2012 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Dict, isDict } from './primitives';
+import { CCITTFaxDecoder } from './ccitt';
+import { DecodeStream } from './stream';
+import { shadow } from '../shared/util';
+
+var CCITTFaxStream = (function CCITTFaxStreamClosure() {
+  function CCITTFaxStream(str, maybeLength, params) {
+    this.str = str;
+    this.dict = str.dict;
+    this.maybeLength = maybeLength;
+    this.params = isDict(params) ? params : Dict.empty;
+
+    DecodeStream.call(this, maybeLength);
+  }
+
+  CCITTFaxStream.prototype = Object.create(DecodeStream.prototype);
+
+  Object.defineProperty(CCITTFaxStream.prototype, 'bytes', {
+    get() {
+      // If this.maybeLength is null, we'll get the entire stream.
+      return shadow(this, 'bytes', this.str.getBytes(this.maybeLength));
+    },
+    configurable: true,
+  });
+
+  CCITTFaxStream.prototype.ensureBuffer = function() {
+    if (this.bufferLength) {
+      return;
+    }
+
+    let ccittFaxDecoder = new CCITTFaxDecoder({
+      K: this.params.get('K'),
+      EndOfLine: this.params.get('EndOfLine'),
+      EncodedByteAlign: this.params.get('EncodedByteAlign'),
+      Columns: this.params.get('Columns'),
+      Rows: this.params.get('Rows'),
+      EndOfBlock: this.params.get('EndOfBlock'),
+      BlackIs1: this.params.get('BlackIs1'),
+    });
+    let data = ccittFaxDecoder.parse(this.bytes, this.maybeLength);
+
+    this.buffer = data;
+    this.bufferLength = data.length;
+    this.eof = true;
+  };
+
+  CCITTFaxStream.prototype.readBlock = function() {
+    this.ensureBuffer();
```

Comment:
Let's make this method as in previous version for now.

Reply:
I've reverted (most of) this, with the exception of the `lookChar` -> `readNextChar` rename and the handling of `-1` return values.